server/download.go

```diff
@@ -446,7 +447,7 @@ func watchDelta(ctx context.Context, g *LimitGroup, c *atomic.Int64, limit int64
 				continue
 			} else if maxDelta > 0 {
 				x := delta / maxDelta
-				if x < 1.2 {
+				if x < 1 {
 					continue
 				}
 
```

Comment:
nit: we might want to set limit to `min(limit + x, numDownloadParts) ?

Reply:
Also, perhaps we want to double? x can be quite large (e.g. below where delta= 147MB and max_delta = 5.5KB):

```
time=2024-03-08T01:19:48.848Z level=DEBUG source=download.go:443 msg="" limit=16 delta="61 MB" max_delta="5.5 KB"
time=2024-03-08T01:19:49.845Z level=DEBUG source=download.go:443 msg="" limit=16 delta="147 MB" max_delta="5.5 KB"
```

Reply:
`*=2` might increase it too much. this should ideally be linear since we don't want to skip too many steps

Reply:
Yeah I was thinking that too